runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/TemplatesSection/TemplatesSection.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC, useState, useMemo, useEffect } from 'react';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import Typography from '#src-landing/components/Typography';
+
+import Categories from './Categories';
+import SearchBar from './SearchBar';
+import styles from './TemplatesSection.module.scss';
+import { categoriesNames, templates } from './TemplatesSection.utils';
+import TemplateTile from './TemplateTile';
+
+const TemplatesSection: FC = () => {
+    const { translate } = useTranslations();
+    const [selectedCategory, setSelectedCategory] = useState<string>(categoriesNames.CATEGORY_ALL);
+    const [searchValue, setSearchValue] = useState<string>('');
+    const [tiles, setTiles] = useState<JSX.Element[]>([]);
+    const [maxDisplayedTiles, setMaxDisplayedTiles] = useState<number>(6);
+
+    const filteredTemplates = useMemo(() => templates
+        .filter(template => selectedCategory === 'All' ? true : template.categories.includes(selectedCategory))
+        .filter(template => template.title.toLowerCase().includes(searchValue.toLowerCase()) 
+            || template.description.toLowerCase().includes(searchValue.toLowerCase())), [selectedCategory, searchValue]);
+
+    const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSearchValue(event.target.value);
+    };
+
+    const handleCategoryChange = (categoryKey: string) => {
+        setSelectedCategory(categoryKey);
+    };
+    
+    const handleResize = () => {
```

Comment:
Utalize CSS variables and media queries, look at the carousel for reference. Timeout is a bit of an hack in my opinion, but clever one

Reply:
thanks, took the carousel for an example. Workin and lookin' good

Reply:
done

Reply:
done, a little bit in my wya. Hope its all right